src/extension/tags.c

```diff
@@ -699,6 +727,136 @@ void _set_runtime_family()
     }
 }
 
+static void _add_custom_event_keyval(zend_array *nonnull meta_ht,
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    zend_string *nonnull name, zend_string *nonnull key,
+    zend_string *nonnull value, bool copy)
+{
+    size_t final_len = LSTRLEN(DD_APPSEC_EVENTS_PREFIX) + ZSTR_LEN(name) +
+                       LSTRLEN(".") + ZSTR_LEN(key);
+
+    smart_str key_str = {0};
+    smart_str_alloc(&key_str, final_len, 0);
+    smart_str_appendl_ex(&key_str, LSTRARG(DD_APPSEC_EVENTS_PREFIX), 0);
+    smart_str_append_ex(&key_str, name, 0);
+    smart_str_appendc_ex(&key_str, '.', 0);
+    smart_str_append_ex(&key_str, key, 0);
+    smart_str_0(&key_str);
+
+    _add_new_zstr_to_meta(meta_ht, key_str.s, value, copy);
+    smart_str_free(&key_str);
+}
+
+static PHP_FUNCTION(datadog_appsec_track_user_login_event)
+{
+    UNUSED(return_value);
```

Comment:
I'd change the returns with `RETVAL_NULL()`

Reply:
I debated this as well, but I've seen it used both ways, although bear in mind that this function doesn't have a return value so it might not actually make a difference.